### DISCUSSION TITLE
course-1-lab question

### DISCUSSION THREAD

**Discussion by gary13139948** [bol-edu/HLS-SOC-Discussions#8] [Q&A] [answered]
Hi,
I have been doing the Lab 1, which is the "Tool Usage". I have stirctly followed the workbook, but in the the end, I get 0 for all output. What problem may I have encountered and how can I deal with it? Thx
![image](https://user-images.githubusercontent.com/108223772/190156149-bdaf8b41-e587-4e5b-abec-d17f72565533.png)

**Comment by Antarcticman** [comment]
Hi,
Is your C Synthesis output correct in Vitis?

**Reply by gary13139948** [reply]
Here is my C Simulation and C Synthesis output, I think it is correct?
<img width="960" alt="image" src="https://user-images.githubusercontent.com/108223772/190400406-91318c8d-a678-4dea-84a8-3521c4db7f10.png">
<img width="959" alt="image" src="https://user-images.githubusercontent.com/108223772/190400575-84db4d01-8d12-4789-9be1-08cb2a2eeff7.png">

**Comment by poppin-mice** [comment, marked as answer]
I met the same problem with you. However, I found that I forgot to restore the directives before the last synthesis and export RTL, The attachment is the instruction inside the workbook. After restoring the directives, it works finally!

![擷取](https://user-images.githubusercontent.com/11053589/190432364-68b27442-4387-44e9-b0b9-a542f6f3f8b5.PNG)

**Reply by gary13139948** [reply]
Thanks for help!! I did it successfully.
